src/app/view/perspectives/chart/chart-data/chart-data.component.ts

```diff
@@ -59,38 +89,78 @@ export class ChartDataComponent implements OnChanges {
   @Input()
   public config: ChartConfig;
 
-  @Input()
-  public resize: Subject<void>;
-
   @Output()
   public patchData = new EventEmitter<DocumentModel>();
 
-  public chartData$ = new BehaviorSubject<ChartData>(null);
+  @ViewChild(ChartVisualizerComponent)
+  public chartVisualizerComponent: ChartVisualizerComponent;
+
+  public chartData$: Observable<ChartData>;
+  public dataSubject = new Subject<Data>();
 
   constructor(private chartDataConverter: ChartDataConverter) {}
 
-  public ngOnChanges(changes: SimpleChanges) {
+  public ngOnInit() {
+    const closingNotifier = this.dataSubject.pipe(debounceTime(100));
+    this.chartData$ = this.dataSubject.pipe(
+      buffer(closingNotifier),
```

Comment:
It seems like this is just a more complicated alternative to using `debounceTime(100)` directly in this pipe instead of `buffer(closingNotifier)`. Or is there any difference?

Reply:
I need to know about each change.

Reply:
Imagine this situation: 
First comes change which indicates that whole chart needs to be updated and then comes another change which indicates about type change. Result is that I need to update whole chart data..